eng/pipelines/stages/build-azure-package-ci-stage.yml

```diff
@@ -0,0 +1,162 @@
+#
+# This template defines a stage named 'build_azure_package_stage' that
+# can be depended on by downstream stages.
+
+parameters:
+
+  # The name of the Abstractions pipeline artifact to download.
+  #
+  # This is used when the referenceType is 'Package'.
+  - name: abstractionsArtifactName
+    type: string
+    default: Abstractions.Artifact
+
+  # The Abstractions package verion to depend on.
+  #
+  # This is used when the referenceType is 'Package'.
+  - name: abstractionsPackageVersion
+    type: string
+
+  # The name of the pipeline artifact to publish.
+  - name: azureArtifactName
+    type: string
+    default: Azure.Artifact
+
+  # The version to apply to the NuGet package and DLLs.
+  - name: azurePackageVersion
+    type: string
+
+  # The type of build to produce (Release or Debug)
+  - name: buildConfiguration
+    type: string
+    default: Release
+    values:
+    - Release
+    - Debug
+
+  # The stages we depend on, if any.
+  - name: dependsOn
+    type: object
+    default: []
+  
+  # The reference type to use:
+  #
+  # Project - dependent projects are referenced directly.
+  # Package - dependent projects are referenced via NuGet packages.
+  #
+  - name: referenceType
+    type: string
+    values:
+    - Package
+    - Project
+
+  # The dotnet CLI verbosity to use.
+  - name: verbosity
+    type: string
+    default: normal
+    values:
+    - quiet
+    - minimal
+    - normal
+    - detailed
+    - diagnostic
+
+stages:
+
+  - stage: build_azure_package_stage
+    displayName: Build Azure Package
+    
+    dependsOn: ${{ parameters.dependsOn }}
+    
+    jobs:
+
+      # ------------------------------------------------------------------------
+      # Build and test on Linux.
+
+      - template: ../jobs/test-azure-package-ci-job.yml@self
+        parameters:
+          abstractionsArtifactName: ${{ parameters.abstractionsArtifactName }}
+          abstractionsPackageVersion: ${{ parameters.abstractionsPackageVersion }}
+          buildConfiguration: ${{ parameters.buildConfiguration }}
+          displayNamePrefix: Linux
+          jobNameSuffix: linux
+          netFrameworkRuntimes: []
+          netRuntimes: [net8.0, net9.0]
+          poolName: Azure Pipelines
+          referenceType: ${{ parameters.referenceType }}
+          verbosity: ${{ parameters.verbosity }}
+          vmImage: ubuntu-latest
+
+      # ------------------------------------------------------------------------
+      # Build and test on Windows
+
+      - template: ../jobs/test-azure-package-ci-job.yml@self
+        parameters:
+          abstractionsArtifactName: ${{ parameters.abstractionsArtifactName }}
+          abstractionsPackageVersion: ${{ parameters.abstractionsPackageVersion }}
+          buildConfiguration: ${{ parameters.buildConfiguration }}
+          displayNamePrefix: Win
+          jobNameSuffix: windows
+          netFrameworkRuntimes: [net462, net47, net471, net472, net48, net481]
```

Comment:
Do we really need to test all those combinations? Don't other jobs just target the lowest version we support?